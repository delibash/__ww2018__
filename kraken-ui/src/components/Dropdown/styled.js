import styled from 'styled-components';
import { colors } from './../../global/constants';

const StyledDropdown = styled.div`
  border: .1rem solid ${colors.GRAY};
  border-radius: .3rem;
  display: flex;
  align-items: center;
  position: relative;
  cursor: pointer;
  i {
    position: absolute;
    height: 2rem;
    right: .8rem;
  }
  select {
    cursor: pointer;
    width: 100%;
    background: none;
    min-height: 3.6rem;
    padding: .2rem .8rem;
    padding-right: 4rem;
    font-size: 1.3rem;
    appearance: none;
    border: 0 none;
  }
`;

export {
  StyledDropdown
};