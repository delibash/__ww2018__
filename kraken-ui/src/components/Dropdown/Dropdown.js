import React from 'react';
import PropTypes from 'prop-types';
import InlineSVG from 'svg-inline-react';
import { StyledDropdown } from './styled';
import { SVGIcons } from './../../global/constants';

const DropdownOption = ({value, label}) => <option value={value}>{label}</option>;

const Dropdown = ({options, ...props}) => {
  return (
    <StyledDropdown>
      <select
        onChange={props.onChange}
      >
        {options && options.map((opt, i) => <DropdownOption key={i} {...opt} />)}
      </select>
      <InlineSVG src={SVGIcons.arrowDown} />
    </StyledDropdown>
  );
};

Dropdown.propTypes = {
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.bool
  ]))
};

DropdownOption.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
  label: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ])
};

export default Dropdown;
