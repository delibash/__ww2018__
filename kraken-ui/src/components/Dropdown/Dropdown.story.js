import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Dropdown from './Dropdown';

storiesOf('Dropdown', module)
  .add('Default Dropdown', () => (
    <Dropdown onChange={action('on-change')} />
  ));
