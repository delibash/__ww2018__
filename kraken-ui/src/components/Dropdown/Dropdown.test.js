import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Dropdown from './Dropdown';

test('Default Button', () => {
  const tree = renderer.create(<Dropdown onChange={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});