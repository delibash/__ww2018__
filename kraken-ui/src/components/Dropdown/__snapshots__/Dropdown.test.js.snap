// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button 1`] = `
.c0 {
  border: .1rem solid rgba(0,0,0,.25);
  border-radius: .3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
}

.c0 i {
  position: absolute;
  height: 2rem;
  right: .8rem;
}

.c0 select {
  cursor: pointer;
  width: 100%;
  background: none;
  min-height: 3.6rem;
  padding: .2rem .8rem;
  padding-right: 4rem;
  font-size: 1.3rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0 none;
}

<div
  className="c0"
>
  <select
    onChange={[Function]}
  />
  <i
    dangerouslySetInnerHTML={
      Object {
        "__html": "
  <svg
    viewBox=\\"0 0 20 20\\"
    height=\\"20\\"
    width=\\"20\\"
    fill=\\"black\\"
  >
    <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\" />
  </svg>
  ",
      }
    }
    src={null}
  />
</div>
`;
